main/src/main/scala/sbt/Cross.scala

```diff
@@ -261,6 +266,14 @@ object Cross {
 
     type ScalaVersion = String
 
+    def resolveNightly(major: String): String = {
+      val url =
+        s"https://raw.githubusercontent.com/scala/community-builds/$major.x/nightly.properties"
+      val props = new java.util.Properties
+      props.load(new java.net.URL(url).openStream)
+      props.getProperty("nightly").ensuring(_ != null)
+    }
```

Comment:
It's highly unusual for the Scala version to be determined by a command that directly goes out to the Internet.
I think there should be an ample logging around here. Like in `-client` I say `"entering *experimental* thin client - BEEP WHIRR"` - https://github.com/sbt/sbt/blob/v1.3.8/main-command/src/main/scala/sbt/internal/client/NetworkClient.scala#L194

Log suggestion:

```
*experimental* nightly cross build - BEEP WHIRR
resolving nightly Scala version for $major from GitHub scala/community-builds...
2.13.2-bin-7da8ce2 found!
```

Reply:
Sounds good.

I think it might also benefit from some caching -- e.g. a cache that times out after 10 minutes -- and thinking about the unhappy path when it can't be resolved, which is now mostly ignored.